contracts/token/ERC721/ERC721.sol

```diff
@@ -21,24 +21,26 @@ abstract contract ERC721 is
      * @inheritdoc ERC721BaseInternal
      */
     function _handleApproveMessageValue(
-        address,
-        uint256,
+        address operator,
+        uint256 tokenId,
         uint256 value
     ) internal virtual override {
         require(value == 0, 'ERC721: payable approve calls not supported');
+        super._handleApproveMessageValue(operator, tokenId, value);
```

Comment:
Why super.? It seems the implementation in ERC721BaseInternal is empty. 

Please explain.

Reply:
If multiple contracts inherit from `ERC721Base` and override this function, they all must call `super` to ensure that all override code is run.

Reply:
But does ERC721Base need `super` in itself? The contract it is inheriting from has no implementation.

Reply:
Not sure what you mean.  `ERC721Base` does not call `super`.  In general, the idea is that someone can import `ERC721BaseInternal` as well as another contract that overrides the hook, and both must call `super` as a convention to ensure that all implementations are included, regardless of inheritance order.  It's not particularly useful here because the hook will be rarely used, but I chose this approach because it supports `payable` ERC721 functions (as specified in the standard) using the preexisting `_beforeTokenTransfer` hook pattern.

Reply:
Thank you for the explanation.